libraries/Microsoft.Bot.Builder.Dialogs.Declarative/Resources/ResourceExplorer.cs

```diff
@@ -32,6 +32,17 @@ public ResourceExplorer()
         {
         }
 
+        public ResourceExplorer(ResourceExplorer source, Func<IResourceProvider, IResourceProvider> transform)
+        {
+            foreach (var resourceProvider in source.resourceProviders.Select(transform))
+            {
+                if (resourceProvider != null)
+                {
+                    AddResourceProvider(resourceProvider);
+                }
+            }
+        }
```

Comment:
I don't understand what you are doing here.

Reply:
refactored out of the SDK as extension method